src/pages/index.js

```diff
@@ -109,21 +181,97 @@ function closeModal(modal) {
   modal.removeEventListener("mousedown", handleOverLayClose);
 }
 
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+
+  api
+    .handleLikeStatus(id, isLiked)
+    .then((card) => {
+      if (card) {
+        evt.target.classList.toggle("card__like-btn_liked");
+        localStorage.setItem(`card-${id}-isLiked`, !isLiked);
+      }
+    })
+    .catch(console.error);
+}
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileNameEl.textContent = nameInput.value;
-  profileDescriptionEl.textContent = descriptionInput.value;
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+
   closeModal(editModal);
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardEl = getCardElement(inputValues);
-  cardList.prepend(cardEl);
-  evt.target.reset();
-  disableButton(cardSubmitBtn, validationConfig);
-  closeModal(cardModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .addCard({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardList.prepend(cardEl);
+      evt.target.reset();
+      disableButton(cardSubmitBtn, validationConfig);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      evt.target.reset();
+      closeModal(avatarModal);
+      disableButton(avatarSubmitBtn, validationConfig);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  console.log(selectedCardId);
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setDeleteText(submitBtn, true);
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setDeleteText(submitBtn, false);
+    });
+
+  closeModal(deleteModal);
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**